packages/@aws-cdk/cloudformation-diff/lib/format-table.ts

```diff
@@ -0,0 +1,111 @@
+import colors = require('colors/safe');
+import stringWidth = require('string-width');
+import table = require('table');
+
+/**
+ * Render a two-dimensional array to a visually attractive table
+ *
+ * First row is considered the table header.
+ */
+export function formatTable(cells: string[][], columns: number | undefined): string {
+  return table.table(cells, {
+    border: TABLE_BORDER_CHARACTERS,
+    columns: buildColumnConfig(calculcateColumnWidths(cells, columns)),
+    drawHorizontalLine: (line) => {
+      // Numbering like this: [line 0] [header = row[0]] [line 1] [row 1] [line 2] [content 2] [line 3]
+      return (line < 2 || line === cells.length) || lineBetween(cells[line - 1], cells[line]);
+    }
+  }).trimRight();
+}
+
+/**
+ * Whether we should draw a line between two rows
+ *
+ * Draw horizontal line if 2nd column values are different.
+ */
+function lineBetween(rowA: string[], rowB: string[]) {
+  return rowA[1] !== rowB[1];
+}
+
+function buildColumnConfig(widths: Array<number | undefined>): { [index: number]: table.ColumnConfig } {
+  const ret: { [index: number]: table.ColumnConfig } = {};
+  widths.forEach((width, i) => {
+    ret[i] = { width, useWordWrap: true } as any; // 'useWordWrap' is not in @types/table
+    if (width === undefined) {
+      delete ret[i].width;
```

Comment:
This `undefined` terminal width has penetrated the logic everywhere. Can we not switch it on/off in a simpler way to avoid `delete ret[i].width` and `Array<number | undefined>`? It seems to have a single source: `formatTable(_, columns: number | undefined)`.